### PR TITLE
Be explicit with tuples for %s formatting

### LINKED ISSUE #1633: `BrokerMetadata` throwing TypeError using old-style string conversion
I was accidentally passing a `BrokerMetada` object rather than a `node_id`, but debugging was more cryptic than it needed to be:

```/Users/jeffwidman/Code/kafka-python/kafka/client_async.py in _maybe_connect(self, node_id)
    354
    355             if conn is None:
--> 356                 assert broker, 'Broker id %s not in current metadata' % node_id
    357
    358                 log.debug("Initiating connection to node %s at %s:%s",

TypeError: not all arguments converted during string formatting
```

Looks like the old-style `%s` is having problems but using `format()` doesn't... 

```ipdb> node_id
BrokerMetadata(nodeId=3, host=u'kafka03.stg.local', port=9092, rack=None)
ipdb> str(node_id)
"BrokerMetadata(nodeId=3, host=u'kafka03.stg.local', port=9092, rack=None)"
ipdb> assert False, "%s" % node_id
*** TypeError: not all arguments converted during string formatting
ipdb> assert False, "{}".format(node_id)
*** AssertionError: BrokerMetadata(nodeId=3, host=u'kafka03.stg.local', port=9092, rack=None)
```

### PR DESCRIPTION
**Pull request by jeffwidman**
Fix #1633

I think I got them all... I would appreciate another pair of eyes to review that I didn't accidentally make mistakes on this.

This is a bit hideous, and definitely overkill, but it's faster than trying to debug `TypeError`s in production...

Alternatively, could switch these to `.format()`. 

Unfortunate that the lovely new python 3.7 syntax f-strings are not backport-able.

---
This change is [Reviewable](https://reviewable.io/reviews/dpkp/kafka-python/1634)

### CODE REVIEW

File: kafka/metrics/stats/sampled_stat.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def purge_obsolete_samples(self, config, now):
                 sample.reset(now)
 
     def _advance(self, config, time_ms):
-        self._current = (self._current + 1) % config.samples
+        self._current = (self._current + 1) % (config.samples,)
```
Comment (tvoinarovskyi) [inline kafka/metrics/stats/sampled_stat.py]: This one's not formatting

**Comment by tvoinarovskyi** [comment]
1 place failed eye test

**Comment by dpkp** [comment]
Looks fine w/ the one modulo issue taras noted. I think `format` is a better long term solution though.

**Comment by jeffwidman** [comment]
Good catch @tvoinarovskyi, I fixed, can you take one last look?

**Pull request merged**